codecs/src/main/scala/Encoder.scala

```diff
@@ -1,22 +1,43 @@
 package com.jpmc.codecs
 
 import com.jpmc.json.Json
+import com.jpmc.json.Json.{JsonArray, JsonObj}
 
 sealed trait JsonEncoder[A] {
   def encode(value: A): Json
 }
 
 object JsonEncoder {
 
-  def apply[A] = ???
-  def from[A] = ???
+  def apply[A](implicit encoder: JsonEncoder[A]): JsonEncoder[A] = encoder
+  def from[A](f: A => Json): JsonEncoder[A] = new JsonEncoder[A] {
+    override def encode(value: A): Json = f(value)
```

Comment:
Could this be converted into a single abstract method? (I vaguely remember seeing a similar refactoring in our live coding sessions). I'm not trying to point out short-comings with this question by the way, I'm just interested if it's possible and thought I would ask someone about this.